src/main/java/picard/fingerprint/FingerprintChecker.java

```diff
@@ -454,30 +473,47 @@ public Map<FingerprintIdDetails, Fingerprint> fingerprintVcf(final Path vcfFile)
     };
 
     /**
-     * Does same thing as {@link #fingerprintSamFile(Path, Function)} but in the old way that required that you pass in the loci of interest.
+     * Does same thing as {@link #fingerprintSamFile(Path, Path, boolean, Function)} but in the old way that required that you pass in the loci of interest.
      * Since the loci are always the same as in {@link #haplotypes} there's no need for this method signature anymore
      *
-     * @deprecated use {@link #fingerprintSamFile(Path, Function)} instead.
+     * @deprecated use {@link #fingerprintSamFile(Path, Path, boolean, Function)} instead.
      */
     @Deprecated
     public Map<FingerprintIdDetails, Fingerprint> fingerprintSamFile(final Path samFile, final IntervalList loci) {
-        return fingerprintSamFile(samFile, HaplotypeProbabilitiesFromSequence::new);
+        return fingerprintSamFile(samFile, null, false, HaplotypeProbabilitiesFromSequence::new);
+    }
+
+    private SamReader getSamReader(final Path samFile, final Path indexPath, final boolean forceIndex) {
+        final SamInputResource samResource = SamInputResource.of(samFile);
+        if (indexPath != null) {
+            // Use of seekableChannelFunction here avoids issue: https://github.com/broadinstitute/picard/issues/1175
+            samResource.index(indexPath, seekableChannelFunction);
```

Comment:
talked about this offline, but it looks like when using a SamInputResource, you need to explicitly give it an index if you want one to be used.  So you need to find the index if you haven't been given it explicitly.  See, as an example, the SamReader open method which takes a Path.  Would also be good to add a test to confirm that the index is used when it is available if not explicitly given as input.

Reply:
Good catch! Fixed and also added a test that should address this.